android/app/src/main/java/app/organicmaps/downloader/DownloaderNotifier.java

```diff
@@ -119,18 +120,22 @@ public Notification buildProgressNotification(@Nullable String countryId, int ma
   }
 
   @NonNull
-  private NotificationCompat.Builder startNotification(@Nullable String countryId)
+  private NotificationCompat.Builder getNotificationBuilder(@Nullable String countryId)
   {
-    final String title = mContext.getString(R.string.app_name);
-
-    return new NotificationCompat.Builder(mContext, CHANNEL_ID)
-        .setAutoCancel(true)
-        .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
-        .setSmallIcon(R.drawable.ic_splash)
-        .setColor(ContextCompat.getColor(mContext, R.color.notification))
-        .setShowWhen(true)
-        .setContentTitle(title)
-        .setContentIntent(getNotificationPendingIntent(countryId));
+    if (!Objects.equals(countryId, mNotificationCountryId))
+    {
+      mNotificationCountryId = countryId;
+      mProgressNotificationBuilder = new NotificationCompat.Builder(mContext, CHANNEL_ID)
+                                         .setAutoCancel(true)
+                                         .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+                                         .setSmallIcon(R.drawable.ic_splash)
+                                         .setColor(ContextCompat.getColor(mContext, R.color.notification))
+                                         .setShowWhen(true)
+                                         .setContentTitle(mContext.getString(R.string.app_name))
+                                         .setContentIntent(getNotificationPendingIntent(mNotificationCountryId))
+                                         .setSound(null);
+    }
+    return mProgressNotificationBuilder;
```

Comment:
Too hard to understand
https://www.franciscomoretti.com/blog/avoid-negative-conditionals

```suggestion
    if (Objects.equals(countryId, mNotificationCountryId))
       return mProgressNotificationBuilder;

    mNotificationCountryId = countryId;
    mProgressNotificationBuilder = new NotificationCompat.Builder(mContext, CHANNEL_ID)
                                       .setAutoCancel(true)
                                       .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
                                       .setSmallIcon(R.drawable.ic_splash)
                                       .setColor(ContextCompat.getColor(mContext, R.color.notification))
                                       .setShowWhen(true)
                                       .setContentTitle(mContext.getString(R.string.app_name))
                                       .setContentIntent(getNotificationPendingIntent(mNotificationCountryId))
                                       .setSound(null);
  return mProgressNotificationBuilder;
```